### DISCUSSION TITLE
onChange converts to controlled component? (Question)

### DISCUSSION THREAD

**Discussion by chiester** [vazco/uniforms#1017] [Q&A] [answered]
When I add an `onChange` prop to my AutoField and then select an item (it's a select list), the selected item no longer stays selected. What's the preferred way for me to be able to use the selected value in React state somehow but still have it be visibly selected and usable by uniforms?

**Comment by wadamek65** [comment, marked as answer]
Hi @chiester ,

If you only need the fields value I think `useField` hook could help you with this. [You can find the documentation for this here](https://uniforms.tools/docs/api-context-data#accessing-context-data). This hook will give you access to all of the fields data and you won't need to pass a `value` and an `onChange` to it explicitly.

**Reply by radekmie** [reply]
Or you can use `useForm` and access the `model` directly. In other words, you should always rely on the current model, not the `onChange` of a field.
